added default tab for docking area

ShahFragcolor opened the pull request:
- added a default tab parameter for docking area to dictate the tab open
- added help text for docking area


<!--
Before submitting your PR, please review the following checklist:

- [ ] **DO NOT** submit a PR without having discussed the change first in a related issue. If none exist, create one first.
- [ ] **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] **DO** keep pull requests small so they can be easily reviewed.
- [ ] **DO** make sure all unit tests pass.
- [ ] **DO** make sure not to introduce any new compiler warnings.
- [ ] **AVOID** breaking the continuous integration build.
- [ ] **AVOID** making significant changes to the overall architecture.
-->

ShahFragcolor committed:
added default tab for docking area

- added a default tab parameter for docking area to dictate the tab open
- added help text for docking area

guusw commented on shards/egui/src/containers/docking.rs:
```diff
@@ -241,23 +251,23 @@ impl LegacyShard for DockArea {
 
   fn outputHelp(&mut self) -> OptionalString {
-    OptionalString(shccstr!("TODO."))
+    OptionalString(shccstr!("This shard always outputs None."))
```
Passthrough I think

guusw replied:
We have a helper
```rs
fn outputHelp(&mut self) -> OptionalString {
    *HELP_OUTPUT_EQUAL_INPUT
  }
```

guusw reviewed (changes requested):
Other than that one comment: good!

guusw commented on shards/egui/src/containers/docking.rs:
```diff
@@ -51,7 +54,13 @@ lazy_static! {
+  (
+    cstr!("DefaultTab"),
+    cstr!("The index of the tab to open by default. 0 being the right most tab."),
+    &INT_OR_NONE_TYPES_SLICE[..],
```
You can just put INT_TYPES (without none), since you're setting the default to 0

ShahFragcolor committed:
edited output helptext

ShahFragcolor committed:
made edits requested

- changed the output help to use the helper instead

- changed int or none to int types
